flows/interfaces.go

```diff
@@ -163,12 +163,17 @@ type Context interface {
 }
 
 type Event interface {
-	CreatedOn() *time.Time
+	CreatedOn() time.Time
 	SetCreatedOn(time.Time)
 
 	Step() StepUUID
 	SetStep(StepUUID)
 
+	FromCaller() bool
+	SetFromCaller(bool)
```

Comment:
Ok am merging this as is for now because I need to take care of some other stuff, and have created #56